src/Microsoft.Azure.SignalR/EndpointProvider/ServiceEndpointManager.cs

```diff
@@ -60,6 +60,12 @@ private void OnChange(ServiceOptions options)
 
             var updatedEndpoints = GetChangedEndpoints(Endpoints);
 
+            await RenameSerivceEndpoints(updatedEndpoints.RenamedEndpoints);
```

Comment:
can it throw and affect the below call? does throw matter?

Reply:
Basically, rename/add/remove are separate, I think we can try-catch inside with some logs for exceptions to ensure each operation won't affect each other.